Src/BootCamp.Chapter/CleanTable.cs

```diff
@@ -0,0 +1,171 @@
+
+            for (var i = 0; i < fixedBalanceLines.Length; i++)
+            {
+                var balances = fixedBalanceLines[i].Split(balanceSeperator);
+                var balanceLine = new StringBuilder(balances[0]);
+
+                for (var j = 1; j < balances.Length; j++)
+                {
+                    balanceLine.Append(", ");
+                    balanceLine.Append(balances[j].Replace(DollarSign.ToString(), ""));
+                }
+
+                balanceLinesForTextTable[i] = balanceLine.ToString();
+            }
+
+            return balanceLinesForTextTable;
+        }
+
+        private void FixCorruptedBalanceFile(string[] corruptedBalanceLines, string[] fixedBalanceLines)
+        {
+            var buildLines = new StringBuilder();
+            var skipCarriageReturn = corruptedBalanceLines.Length - 1;
+
+            for (var i = 0; i < corruptedBalanceLines.Length; i++)
+            {
+                fixedBalanceLines[i] = corruptedBalanceLines[i].Replace("_", "");
+
+                if (!isValidLine(fixedBalanceLines[i]))
+                {
+                    throw new Exception("Invalid Balance.");
+                }
+
+                buildLines.Append(fixedBalanceLines[i]);
+
+                if (i < skipCarriageReturn)
+                {
+                    buildLines.Append("\r\n");
+                }
+            }
+
+            File.WriteAllText(_fixedBalanceFile, buildLines.ToString());
+        }
+
+        private bool isValidLine(string line)
+        {
+            var words = line.Split(",");
+
+            if (!IsValidName(words[0]))
+            {
+                throw new InvalidBalancesException($"Invalid Name {words[0]}");
+            }
+
+            for (var i = 1; i < words.Length; i++)
+            {
+                if (!IsValidBalance(words[i]))
+                {
+                    throw new InvalidBalancesException($"Invalid Balance {words[i]}");
+                }
+            }
+
+            return true;
+        }
+
+        private bool IsValidName(string name)
+        {
+            for (var i = 0; i < name.Length; i++)
+            {
+                if (!IsValidLetter(name[i]) && !HasPeriodAtEnd(name))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private bool IsValidLetter(char letter)
+        {
+            if ((letter >= 'a' && letter <= 'z') ||
+                (letter >= 'A' && letter <= 'Z') ||
+                (letter == ' ') ||
+                (letter == '\'') ||
+                (letter == '-'))
+            {
+                return true;
+            }
+            return false;
+        }
+
+        private bool HasPeriodAtEnd(string name)
+        {
+            if (name[name.Length - 1] == '.')
+            {
+                return true;
+            }
+            return false;
+        }
+
+        private bool IsValidBalance(string balance)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make 'IsValidBalance' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)